src/OrchardCore/OrchardCore.DisplayManagement/ShapeFactoryExtensions.cs

```diff
@@ -0,0 +1,195 @@
+using System.Collections.Concurrent;
+using Castle.DynamicProxy;
+using OrchardCore.DisplayManagement.Shapes;
+using OrchardCore.DisplayManagement.Views;
+
+namespace OrchardCore.DisplayManagement;
+
+public static class ShapeFactoryExtensions
+{
+    private static readonly ConcurrentDictionary<Type, Type> _proxyTypesCache = [];
+    private static readonly ProxyGenerator _proxyGenerator = new();
+    private static readonly Func<ValueTask<IShape>> _newShape = () => ValueTask.FromResult<IShape>(new Shape());
+
+    /// <summary>
+    /// Creates a new generic shape by copying the properties of an object.
+    /// </summary>
+    /// <param name="factory">The <see cref="IShapeFactory"/>.</param>
+    /// <param name="shapeType">The type of shape to create.</param>
+    /// <param name="model">The model to copy.</param>
+    /// <returns></returns>
+    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, string shapeType, TModel model)
+    {
+        return factory.CreateAsync(shapeType, Arguments.From(model));
+    }
+
+    /// <summary>
+    /// Creates a new generic shape instance.
+    /// </summary>
+    public static ValueTask<IShape> CreateAsync(this IShapeFactory factory, string shapeType)
+    {
+        return factory.CreateAsync(shapeType, _newShape);
+    }
+
+    /// <summary>
+    /// Creates a new generic shape instance and initializes it.
+    /// </summary>
+    public static ValueTask<IShape> CreateAsync(this IShapeFactory factory, string shapeType, Func<ValueTask<IShape>> shapeFactory)
+    {
+        return factory.CreateAsync(shapeType, shapeFactory, null, null);
+    }
+
+    /// <summary>
+    /// Creates a dynamic proxy instance for the type and initializes it.
+    /// </summary>
+    /// <typeparam name="TModel">The type to instantiate.</typeparam>
+    /// <param name="factory">The <see cref="IShapeFactory"/>.</param>
+    /// <param name="initialize">The initialization method.</param>
+    /// <returns></returns>
+    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, Action<TModel> initialize = null)
```

Comment:
```suggestion
    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, Action<TModel> initializeAction = null)
```

Reply:
Even though the name is not clear `initializeAction` or `initializationAction is much cleaner

Reply:
I don't like using Action as a prefix. sorry

Reply:
This is a common pattern I saw in the ASP.NET Core team or you could use `initializeMethod`

Reply:
Still not a fan. Again most of the code existed before this PR.